default.nix

```diff
@@ -122,9 +122,27 @@ let
           pkgs.netcat-gnu
           pkgs.nodejs
         ];
+
+        reformatPackageName = pname:
+          let
+            # regex adapted from `validate-npm-package-name`
+            # will produce 3 parts e.g.
+            # "@someorg/somepackage" -> [ "@someorg/" "someorg" "somepackage" ]
+            # "somepackage" -> [ null null "somepackage" ]
+            parts = builtins.tail (builtins.match "^(@([^/]+)/)?([^/]+)$" pname);
+            # if there is no organisation we need to filter out null values.
+            non-null = builtins.filter (x: x != null) parts;
+          in builtins.concatStringsSep "-" non-null;
+
+        # NOTE: this is an import-from-derivation
+        packageJSON = pkgs.lib.importJSON "${toString src}/package.json";
```

Comment:
What about `importJSON (src + "/package.json")`? also I don't think this is always IFD, but only when `src` is not a `path`.

moreover, it'd be nice to do a `pathExists` and, if false, then throw a nice error!

Reply:
good points, I will see if I can improve that.

Reply:
sort of fixed in 502dee9 . It's still falling back on the defaults but emits a warning instead if the package.json is missing.